zkevm-circuits/src/copy_circuit.rs

```diff
@@ -265,25 +280,31 @@ impl<F: Field> SubCircuitConfig<F> for CopyCircuitConfig<F> {
                     );
                 },
             );
+            cb.require_equal(
+                "write value == read value",
+                meta.query_advice(value, Rotation::cur()),
+                meta.query_advice(value, Rotation::next()),
+            );
+            cb.require_equal(
+                "value_acc_rlc is same for read-write rows",
+                meta.query_advice(value_acc_rlc, Rotation::cur()),
+                meta.query_advice(value_acc_rlc, Rotation::next()),
+            );
             cb.condition(
-                not::expr(tag.value_equals(CopyDataType::RlcAcc, Rotation::next())(
-                    meta,
-                )),
+                and::expr([
+                    not::expr(meta.query_advice(is_last, Rotation::next())),
+                    not::expr(meta.query_advice(is_pad, Rotation::cur())),
+                ]),
```

Comment:
This'd be a little bit cheaper
```suggestion
                not::expr(
                    and::expr([
                        meta.query_advice(is_last, Rotation::next()),
                        meta.query_advice(is_pad, Rotation::cur()),
                    ]),
                ),
```

Reply:
fixed in [d2bdc68](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1419/commits/d2bdc68414d218732a56d4e1d038b1c11859c9e6)

Reply:
@han0110 , it should be `or` instead of `and`. Could you help to confirm? fixed in [b696d4a](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1419/commits/b696d4a9bf92904ad71c8234fd21a9986091eb40)
```rust
                not::expr(
                    or::expr([
                        meta.query_advice(is_last, Rotation::next()),
                        meta.query_advice(is_pad, Rotation::cur()),
                    ]),
                ),
```